demisto_sdk/commands/content_graph/objects/pack.py

```diff
@@ -603,3 +603,368 @@ def to_nodes(self) -> Nodes:
             self.to_dict(),
             *[content_item.to_dict() for content_item in self.content_items],
         )
+
+    def enhance_pack_properties(self, marketplace: MarketplaceVersions):
+        """
+        Enhancing the Pack object properties before dumping into a dictionary.
+        - Adding tags considering the pack content items and marketplace.
+        - Replacing the `author` property from XSOAR to XSIAM if the prepare is to marketplacev2.
+        - Getting into the `version_info` property the pipeline_id variable.
+        - Calculating the `server_min_version` by the pack's content items fromversion`.
+
+        Args:
+            marketplace (MarketplaceVersions): The marketplace to which the pack should belong to.
+        """
+        self.tags = self.get_pack_tags(marketplace)
+        self.author = (
+            self.author
+            if marketplace == MarketplaceVersions.XSOAR
+            else self.author.replace("XSOAR", "XSIAM")  # type:ignore[union-attr]
+        )
+        self.version_info = os.environ.get("CI_PIPELINE_ID", "")
+        self.server_min_version = (
+            self.server_min_version
+            or str(
+                max(
+                    (
+                        parse(content_item.fromversion)
+                        for content_item in self.content_items
+                    ),
+                    default=MARKETPLACE_MIN_VERSION,
+                )
+            )
+            or MARKETPLACE_MIN_VERSION
+        )
+
+    def enhance_metadata(self, marketplace: MarketplaceVersions) -> dict:
+        """
+        Enhancing the pack metadata properties after dumping into a dictionary. (properties that can't be calculating before)
```

Comment:
because I prefer not to change the object property value just for the metadata, for example, self.content_items is calculated by the graph/pydantic etc., so I don't think we need to change it because in the metadata it should have a different value